### PR TITLE
Make phan config more strict

### PR DESCRIPTION
**Pull request by MorrisJobke**
Reminder for myself to take this and get it step by step in.

cc @rullzer

### CODE REVIEW

**Comment by MorrisJobke** [comment]
Rebased and resolved quite some stuff to make it a bit easier to dive through them:

Before:
bildschirmfoto 2018-03-07 um 11 43 28

After:
bildschirmfoto 2018-03-07 um 11 43 33

**Comment by rullzer** [comment]
Wow seems we did good :)

**Comment by MorrisJobke** [comment]
1814 warnings to go xD

**Comment by MorrisJobke** [comment]
2018-03-13: 1788

File: build/.phan/config.php [hunk 1]
```diff
@@ -155,7 +201,7 @@
 	// Issue::SEVERITY_CRITICAL. Setting it to only
 	// critical issues is a good place to start on a big
 	// sloppy mature code base.
-	'minimum_severity' => \Phan\Issue::SEVERITY_CRITICAL,
+	'minimum_severity' => \Phan\Issue::SEVERITY_NORMAL,
```
Comment (MorrisJobke) [inline build/.phan/config.php]: Okay this causes ~1200 warnings -> let's do this later

**Comment by MorrisJobke** [comment]
Not the way to go -> closing.

**Pull request closed without merging**